### PR TITLE
Use own phar compiler

### PR DESCRIPTION
**Pull request by ricknox**
### Pull request overview card

| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| License       | MIT

### Pull request description

#2: This pull request removes the [box project](https://github.com/box-project/box2) configuration from the repository and implements a own phar compiler to build new Butler releases.

### CODE REVIEW

File: src/Phar/Compiler.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    /**
+     * Return the version from the Composer configuration.
+     *
+     * @return string
+     */
+    public function getVersion()
+    {
+        $content = file_get_contents(__DIR__ . '/../../composer.json');
```
Comment (greg0ire) [inline src/Phar/Compiler.php]: What happens if this file is invalid? Is there a clear error message?

File: src/Phar/Compiler.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        unset($phar);
+    }
+
+    /**
+     * Add file to phar archive
+     *
+     * @param \Phar $phar
+     * @param \SplFileInfo $file
+     * @param string $root
+     */
```
Comment (greg0ire) [inline src/Phar/Compiler.php]: Why don't you add type hints for string? You could get rid of these comments. Do you need to be compatible with old versions of php?

File: src/Phar/Compiler.php (hunk 1 above)
Comment (ricknox) [reply src/Phar/Compiler.php]: Thank you @greg0ire. This has been fixed in https://github.com/ricknox/butler/pull/3/commits/589e72dcf7efcdaa7438813598e07ee9b42b61ff.

File: src/Phar/Compiler.php (hunk 2 above)
Comment (ricknox) [reply src/Phar/Compiler.php]: Can you advise me in this? When I refactor the code to:

```php
/**
 * Add file to phar archive
 */
protected function addFile(\Phar $phar, \SplFileInfo $file, string $root): void
```

PhpStorm complains about missing PHPDoc arguments. Is that something I can ignore?

Comment (greg0ire) [reply src/Phar/Compiler.php]: I don't use PhpStorm so I wouldn't know how to configure it to ignore arguments, but that's definitely something that *should* be ignored in my opinion, see https://github.com/sonata-project/dev-kit/issues/345 and symfony/symfony#24611

Comment (ricknox) [reply src/Phar/Compiler.php]: Ah. As from the links, from the other repositories, the PHPDoc arguments can be ignored for this pull request.

Thank you @greg0ire. This has been fixed in https://github.com/ricknox/butler/pull/3/commits/bb94ac56e5ee973aca8a4b5dd5cf7dde73f8f97d.

**Pull request merged**